src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -236,14 +231,15 @@ private void subscribeToServerCertificateUpdatesNoCSR(@NonNull GetCertificateReq
 
     private void subscribeToClientCertificateUpdatesNoCSR(@NonNull GetCertificateRequest certificateRequest,
                                                           @NonNull PublicKey publicKey,
-                                                          @NonNull Consumer<X509Certificate[]> cb)
+                                                          @NonNull BiConsumer<X509Certificate[], X509Certificate[]> cb)
             throws CertificateGenerationException {
         CertificateGenerator certificateGenerator =
                 new ClientCertificateGenerator(
                         CertificateHelper.getX500Name(certificateRequest.getServiceName()),
                         publicKey, cb, certificateStore, certificatesConfig, clock);
 
         certExpiryMonitor.addToMonitor(certificateGenerator);
+        certificateGeneratorRegistry.registerGenerator(certificateGenerator);
```

Comment:
What is this weird `certificateGeneratorRegistry` contraption?

Each of the monitors were maintaining a Set of generators that we had to register separately, that didn't make it very ideal when we wanted to add a new generator for things like configuration changes. This `certificateGeneratorRegistry` is basically a wrapper around a `CopyOnWriteSet` that all monitors use, with that in place we can call the generators from different places (config changes, cert expiration, connectivity changed) without having to unnecessarily register them and de register them for each.